vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1463,6 +1473,25 @@ func resourceVSphereComputeClusterApplyVsanConfig(d *schema.ResourceData, meta i
 		return err
 	}
 	version := viapi.ParseVersionFromClient(client)
+
+	if version.AtLeast(viapi.VSphereVersion{Product: version.Product, Major: 8, Minor: 0}) {
+		if !d.Get("vsan_enabled").(bool) && d.Get("vsan_esa_enabled").(bool) {
+			return fmt.Errorf("vSAN ESA service cannot be enabled on cluster due to vSAN is disabled: %s", d.Get("name").(string))
+		}
+		if !d.HasChange("vsan_enabled") && d.HasChange("vsan_esa_enabled") {
+			return fmt.Errorf("vSAN ESA service must be configured along with vSAN service: %s", d.Get("name").(string))
+		}
+		if d.Get("vsan_esa_enabled").(bool) {
+			// need to be revised if GetOkExists() is deprecated in the future
+			unmapEnabled, unmapOK := d.GetOkExists("vsan_unmap_enabled")
```

Comment:
I would not rely on GetOkExists at all. Like i wrote up in my longwinded comment above. I think it's safer to not have this attribute be computed, just use `Get` which means `nil/unset` should be resolved to us as `False` and then error if it's False when they are trying to enable vSAN ESA.

Reply:
> I would not rely on GetOkExists at all. Like i wrote up in my longwinded comment above. I think it's safer to not have this attribute be computed, just use `Get` which means `nil/unset` should be resolved to us as `False` and then error if it's False when they are trying to enable vSAN ESA.

that means user must type "vsan_unmap_enabled = true" along with "vsan_esa_enabled = true" in .tf file, otherwise provider throw error. It avoids such verbosity if-else check, let me change to this way and explain this restrictions clearly in documentation.